src/Components/HorizontalCard/HorizontalProductCard.jsx

```diff
@@ -0,0 +1,141 @@
+        badgeText, 
+        outOfStock, 
+        quantity
+    } = productDetails;
+
+    const [productQuantity, setProductQuantity] = useState(quantity)
+
+    async function onQuantityChange(event)
+    {
+        setProductQuantity(event.target.value)
+        let newQuantity = event.target.value
+        let quantityUpdateResponse = await axios.patch(
+            `https://bookztron.herokuapp.com/api/cart/${_id}`,
+            {
+                newQuantity
+            },
+            {
+                headers:
+                {
+                    'x-access-token': localStorage.getItem('token'),
+                }
+            }
+        )
+
+        if(quantityUpdateResponse.data.status==="ok")
+        {
+            dispatchUserCart({type: "UPDATE_USER_CART",payload: quantityUpdateResponse.data.user.cart})
+            showToast("success","","Item quantity successfully updated!")
+        }
+        else
+        {
+            showToast("error","","Something went wrong!")
+        }
+    }
+
+    async function removeItemFromCart(event)
+    {
+        const token=localStorage.getItem('token')
+
+        if(token)
+        {
+            const user = jwt_decode(token)
+                
+            if(!user)
+            {
+                localStorage.removeItem('token')
+                showToast("warning","","Kindly Login")
+                navigate('/login')
+            }
+            else
+            {
+                let cartUpdateResponse = await axios.delete(
+                    `https://bookztron.herokuapp.com/api/cart/${productDetails._id}`,
+                    {
+                        headers:
+                        {
+                            'x-access-token': localStorage.getItem('token'),
+                        }
+                    },
+                    {
+                        productDetails
+                    }
+                )
+                if(cartUpdateResponse.data.status==="ok")
+                {
+                    dispatchUserCart({type: "UPDATE_USER_CART",payload: cartUpdateResponse.data.user.cart})
+                    showToast("success","","Item successfully deleted from cart")
+                }
+            }
+        }
+        else
+        {
+            showToast("warning","","Kindly Login")
+        } 
+    }
+
+    return (
+        <div className="card-basic-horizontal">
+            <img className="cart-item-book-img" src={imgSrc} alt={imgAlt}/>
+            <div id="cart-item-detail" className="card-item-details">
+
+                <h4 id="item-title">{bookName}</h4>
+                <p className="item-author">- By &nbsp;{author}</p>
+                <p className="price-details">
+                                &#8377; {discountedPrice} &nbsp;&nbsp;<del>&#8377; {originalPrice}</del> &nbsp;&nbsp;
+                    <span className="discount-on-card">{discountPercent}% off</span>
+                </p>
+    
+                <div className="item-cart-quantity">
```

Comment:
just a suggestion but it would be great if there is and add and reduce quantity option in the cart itself.
Also, if 2 items for the same product is added i have to click on remove from cart twice for it to get deleted from my cart.

Reply:
1. I added an input field for it, any quantity change in the input field will reflect in the bill as well.
2. I think this is a better UI, because if I implemented as your suggestion and user needs to remove a product from cart which has 15 quantity, he/she would have to click 15 times to remove it 😅